zamiel-bot.js

```diff
@@ -2369,48 +2414,176 @@ DiscordBot.on('message', (message) => {
         }
     }
 
-    // Discord special info commands
-    if (content.match(/^!roll\b/) || content.match(/^!rand\b/) || content.match(/^!random\b/)) {
-        const m = content.match(/^!\w+ (\d+) (\d+)$/);
+    const args = message.content.slice(1).trim().split(/ +/g);
+    const command = args.shift().toLowerCase();
+
+    switch (command) {
+    case 'roll':
+    case 'rand':
+    case 'random': {
+        const m = msg.match(/^!\w+ (\d+) (\d+)$/);
         let randomMin;
         let randomMax;
         if (m) {
-            randomMin = m[1];
-            randomMax = m[2];
+            randomMin = m['1'];
```

Comment:
what is this

Reply:
new airbnb rule https://gyazo.com/b9c82c599b974c9de5b871b976f8f462

Reply:
the code you wrote is nonsense and is not actually array destructuring.
the correct way to write this is:
```
[ , randomMin, randomMax] = m
```
please fix all the places in the code that you did this

Reply:
That's actually your code not mine, it's your old function.

Reply:
you changed my code from:
```
m[1]
```
to
```
m['1']
```
which is nonsense, because m is an array and not an object.
m has no property called "1".
for more information on the `match` function, see this page: 
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/match

more information on what array destructuring actually is can be found on the eslint documentation:
https://eslint.org/docs/rules/prefer-destructuring